Salehi2025/Figure5.qmd

```diff
@@ -0,0 +1,450 @@
+
+# Perform dummy (one-hot) encoding for selected predictors
+temp_df_one_hot <- model.matrix(~ 0 + 
+                                  sex_num_Men +
+                                  continent + 
+                                  age_category + 
+                                  income_group_HIC + 
+                                  Usage_high,
+                                data = temp_df) %>%
+  as.data.frame()
+
+# Clean column names by replacing spaces and dashes with underscores
+colnames(temp_df_one_hot) <- gsub(" |-", "_", colnames(temp_df_one_hot))
+
+# Store the dummy variable names for later model specification
+dummy_var_names <- colnames(temp_df_one_hot)
+
+# Combine dummy variables with original metadata
+adult_metadata <- cbind(adult_metadata, temp_df_one_hot)
+
+# Clean up temporary dataframes
+rm(temp_df, temp_df_one_hot)
+```
+
+This section defines utility functions used later in the analysis for formatting results and interpreting significance.
+
+```{r, helpers}
+## Helpers
+
+# Format numbers: Convert values below 0.001 to "<0.001"
+to_neat <- function(x, threshold = 0.001) {
+  ifelse(x > threshold, as.character(x), "<0.001")
+}
+
+# Assign significance stars for confidence intervals (log-scale)
+get_stars_CI_log <- function(lower, upper) {
+  ifelse(lower > 1 | upper < 1, "*", "ns")
+}
+
+# Assign significance stars for confidence intervals (linear scale)
+get_stars_CI <- function(lower, upper) {
+  ifelse(lower > 0 | upper < 0, "*", "ns")
+}
+```
+
+# Modeling
+
+First, we define the response variables (ARG_load, shannon_diversity) and income groups (LMIC, HIC) used for stratified statistical modeling.
+
+```{r}
+responses <- c("ARG_load", "shannon_diversity")
+incomes <- 0:1
+set.seed(123123)
+```
+
+This section describes four variants of Bayesian regression models. Each model was run separately for two outcomes: ARG load and Shannon diversity, and for two income groups: HIC (High-Income Countries) and LMIC (Low- and Middle-Income Countries).
+
+Model Variants:
+
+1.  Model 1 (Reference): Fixed effects only. No adjustment for bioproject or sequencing depth.
+
+2.  Model 2: Includes a random intercept for bioproject to account for between-study variability.
+
+3.  Model 3: Adds log-transformed sequencing depth log(readcount) as a covariate.
+
+4.  Model 4: Combines both: a random intercept for bioproject and sequencing depth as a covariate.
+
+Note:
+
+Model fitting is computationally intensive and is therefore performed separately in an external Quarto document, please run that script before the next code chunk: (Figure5models.qmd).
+
+The resulting model objects are saved as `.rds` files (e.g. `fit1.rds`, `fit2.rds`, etc.) and loaded back into the this Quarto document for result summarization and plotting.
+
+```{r, models}
+# Load models
+fit_list1 <- readRDS("fit1.rds")
+fit_list2 <- readRDS("fit2.rds")
+fit_list3 <- readRDS("fit3.rds")
+fit_list4 <- readRDS("fit4.rds")
+

```

Comment:
if possible, put it in its own chunk (not sure if possible?)

Reply:
some more explanation what each different model / summary below does would help the reader to follow